pkg/terminal/command.go

```diff
@@ -2451,15 +2452,26 @@ func (c *Commands) sourceCommand(t *Term, ctx callContext, args string) error {
 		return fmt.Errorf("wrong number of arguments: source <filename>")
 	}
 
+	if args == "-" {
+		return t.starlarkEnv.REPL()
+	}
+
+	if runtime.GOOS != "windows" && strings.HasPrefix(args, "~") {
+		home, err := os.UserHomeDir()
+		if err == nil {
+			if args == "~" {
```

Comment:
Could we simplify this with `strings.Replace(args, "~", home, 1)`.

Reply:
It should only be expanded if it's at the start, for example `blah~` shouldn't be expanded. Also it shouldn't be expanded if it isn't a directory component, like `~blah/blah.star`.